src/components/ExercisesEditPage.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component, PropTypes } from 'react';
+
+export default class ExercisesEditPage extends Component {
+
+  constructor() {
+    super();
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.fetchExercise(this.props.params.id);
+  }
+
+  componentWillUnmount() {
+    this.props.onPageLeave();
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+    const { title, text, category, batch, fileUpload } = this.refs;
+    const { exercise, handleSubmit } = this.props;
+
+    handleSubmit(
+      exercise.get('id'),
+      title.value,
+      text.value,
+      category.value,
+      batch.value,
+      fileUpload.checked
+    );
+  }
+
+  renderOptionItems(set, chosenMember, refName) {
```

Comment:
1. rename `renderOptionItems` to `renderSelect`
2. print field's label
3. `chosenMember` could be simple id as a `defaultValue`